internal/plan/plan.go

```diff
@@ -27,33 +27,36 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
-type Setup struct {
-	Layers []*Layer
+type Plan struct {
+	Layers      []*Layer            `yaml:"-"`
+	Summary     string              `yaml:"summary,omitempty"`
+	Description string              `yaml:"description,omitempty"`
+	Services    map[string]*Service `yaml:"services,omitempty"`
 }
 
-func (s *Setup) AddLayer(layer *Layer) {
-	s.Layers = append(s.Layers, layer)
+type Layer struct {
+	Order       int                 `yaml:"-"`
+	Label       string              `yaml:"-"`
+	Summary     string              `yaml:"summary,omitempty"`
+	Description string              `yaml:"description,omitempty"`
+	Services    map[string]*Service `yaml:"services,omitempty"`
 }
 
-type Layer struct {
-	Order       int
-	Label       string
-	Summary     string
-	Description string
-	Services    map[string]*Service
+func (l *Layer) IsDynamic() bool {
+	return l.Label == ""
```

Comment:
Will it ever make sense to have more than one dynamic layer?  Perhaps one atop and one in the middle?  I guess we might eventually have dynamic layers with labels and orders. Not a big deal either way because we can change this method in a compatible way, but food for thought.